components/org.wso2.carbon.crypto.provider.hsm/src/test/java/org/wso2/carbon/crypto/provider/hsm/cryptoprovider/objecthandlers/KeyHandlerTest.java

```diff
@@ -0,0 +1,204 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.crypto.provider.hsm.cryptoprovider.objecthandlers;
+
+import iaik.pkcs.pkcs11.Mechanism;
+import iaik.pkcs.pkcs11.Session;
+import iaik.pkcs.pkcs11.objects.AESSecretKey;
+import iaik.pkcs.pkcs11.objects.DES2SecretKey;
+import iaik.pkcs.pkcs11.objects.DES3SecretKey;
+import iaik.pkcs.pkcs11.objects.DESSecretKey;
+import iaik.pkcs.pkcs11.objects.Key;
+import iaik.pkcs.pkcs11.objects.PrivateKey;
+import iaik.pkcs.pkcs11.objects.PublicKey;
+import iaik.pkcs.pkcs11.objects.SecretKey;
+import iaik.pkcs.pkcs11.wrapper.PKCS11Constants;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.base.ServerConfigurationException;
+import org.wso2.carbon.crypto.api.CryptoException;
+import org.wso2.carbon.crypto.provider.hsm.TestUtil;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.exception.HSMCryptoException;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.util.KeyTemplateGenerator;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.util.SessionHandler;
+
+public class KeyHandlerTest {
+
+    private SessionHandler sessionHandler;
+    private Session session;
+    private KeyHandler keyHandler;
+    private String keyAliasNotInHSM = "notInsideHSM";
+
+    @BeforeClass
+    public void setUpClass() throws CryptoException, ServerConfigurationException {
+
+        sessionHandler = TestUtil.getSessionHandler();
+    }
+
+    @BeforeMethod
+    public void setUp() throws CryptoException {
+
+        session = sessionHandler.initiateSession(0, null, true);
+        keyHandler = new KeyHandler(session);
+    }
+
+    @AfterMethod
+    public void tearDown() throws CryptoException {
+
+        sessionHandler.closeSession(session);
+    }
+
+    @Test(dataProvider = "sampleKeyRetrievalData")
+    public void testRetrieveKey(Key keyTemplate) {
+
+        try {
+            Key key = keyHandler.retrieveKey(keyTemplate);
+            Assert.assertEquals(String.valueOf(key.getLabel().getCharArrayValue()),
+                    String.valueOf(keyTemplate.getLabel().getCharArrayValue()));
+            Assert.assertTrue(key.getObjectHandle() != -1);
+        } catch (CryptoException e) {
+            if (keyAliasNotInHSM.equals(String.valueOf(keyTemplate.getLabel().getCharArrayValue()))) {
+                Assert.assertEquals(e.getMessage(), String.format("Requested key with key alias %s can't be " +
+                        "found inside the HSM.", String.valueOf(keyTemplate.getLabel().getCharArrayValue())));
+            }
+        }
+    }
+
+    @DataProvider(name = "sampleKeyRetrievalData")
+    public Object[][] getSampleKeyData() {
+
+        ;
```

Comment:
why this ;

Reply:
Removed with PR https://github.com/wso2-extensions/carbon-crypto-provider-hsm/pull/2